sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/DeliveryHandler.java

```diff
@@ -0,0 +1,489 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Tests?

Reply:
yes yes tests are coming :) once we pass the current tests pass.

Reply:
Added tests covering the new types.